code/modules/projectiles/guns/powered.dm

```diff
@@ -0,0 +1,123 @@
+/obj/item/gun/ballistic/automatic/powered
+	mag_type = /obj/item/ammo_box/magazine/gauss
+	can_suppress = FALSE
+
+	var/obj/item/stock_parts/cell/cell
+	var/cell_type = /obj/item/stock_parts/cell/gun
+	var/charge_sections = 3
+
+	var/shaded_charge = FALSE //if this gun uses a stateful charge bar for more detail
+	var/automatic_charge_overlays = TRUE	//Do we handle overlays with base update_icon()?
+
+	var/internal_cell = FALSE ///if the gun's cell cannot be replaced
+	var/small_gun = FALSE ///if the gun is small and can only fit the small gun cell
+	var/big_gun = FALSE ///if the gun is big and can fit the comically large gun cell
+	var/unscrewing_time = 20 ///Time it takes to unscrew the cell
+	var/sound_volume = 40 //Volume of loading/unloading cell sounds
+
+
+/obj/item/gun/ballistic/automatic/powered/Initialize()
+	. = ..()
+	if(cell_type)
+		cell = new cell_type(src)
+	else
+		cell = new(src)
+	update_icon()
+
+/obj/item/gun/ballistic/automatic/powered/examine(mob/user)
+	. = ..()
+	if(cell)
+		. += "<span class='notice'>[src]'s cell is [round(cell.charge / cell.maxcharge, 0.1) * 100]% full.</span>"
+	else
+		. += "<span class='notice'>[src] doesn't seem to have a cell!</span>"
+
+/obj/item/gun/ballistic/automatic/powered/can_shoot()
+	if(QDELETED(cell))
+		return 0
+
+	var/obj/item/ammo_casing/caseless/gauss/shot = chambered
+	if(!shot)
+		return 0
+	if(cell.charge < shot.energy_cost * burst_size)
+		return 0
+	. = ..()
+
+/obj/item/gun/ballistic/automatic/powered/shoot_live_shot(mob/living/user, pointblank = FALSE, mob/pbtarget, message = 1, stam_cost = 0)
+	var/obj/item/ammo_casing/caseless/gauss/shot = chambered
+	cell.use(shot.energy_cost)
+	. = ..()
+
+/obj/item/gun/ballistic/automatic/powered/get_cell()
+	return cell
+
+/obj/item/gun/ballistic/automatic/powered/nopin
+	pin = null
+	spawnwithmagazine = FALSE
+
+//the things below were taken from energy gun code. blame whoever coded this, not me
+/obj/item/gun/ballistic/automatic/powered/attackby(obj/item/A, mob/user, params)
+	if (!internal_cell && istype(A, /obj/item/stock_parts/cell/gun))
+		var/obj/item/stock_parts/cell/gun/C = A
+		if (!cell)
+			insert_cell(user, C)
+	return ..()
+
+/obj/item/gun/ballistic/automatic/powered/proc/insert_cell(mob/user, obj/item/stock_parts/cell/gun/C)
+	if(small_gun && !istype(C, /obj/item/stock_parts/cell/gun/mini))
+		to_chat(user, "<span class='warning'>\The [C] doesn't seem to fit into \the [src]...</span>")
+		return FALSE
+	if(!big_gun && istype(C, /obj/item/stock_parts/cell/gun/large))
+		to_chat(user, "<span class='warning'>\The [C] doesn't seem to fit into \the [src]...</span>")
+		return FALSE
+	if(user.transferItemToLoc(C, src))
+		cell = C
+		to_chat(user, "<span class='notice'>You load the [C] into \the [src].</span>")
+		playsound(src, load_sound, sound_volume, load_sound_vary)
+		update_icon()
+		return TRUE
+	else
+		to_chat(user, "<span class='warning'>You cannot seem to get \the [src] out of your hands!</span>")
```

Comment:
\the is included in [src] im pretty sure

Reply:
for other lines too

Reply:
fixed,along with other lines